src/EzPlatformCoreBundle/bundle/DependencyInjection/Compiler/PlatformShConfigPass.php

```diff
@@ -0,0 +1,180 @@
+                    continue;
+                }
+
+                $container->setParameter('ezplatform.session.handler_id', 'ezplatform.core.session.handler.native_redis');
+                $container->setParameter('ezplatform.session.save_path', sprintf('%s:%d', $endpoint['host'], $endpoint['port']));
+            }
+        }
+
+        if (isset($relationships['solr'])) {
+            foreach ($relationships['solr'] as $endpoint) {
+                if ($endpoint['scheme'] !== 'solr') {
+                    continue;
+                }
+
+                $container->setParameter('search_engine', 'solr');
+                $container->setParameter('solr_dsn', sprintf('http://%s:%d/%s', $endpoint['host'], $endpoint['port'], 'solr'));
+                // To set solr_core parameter we assume path is in form like: "solr/collection1"
+                $container->setParameter('solr_core', substr($endpoint['path'], 5));
+            }
+        }
+
+        // We will pick a varnish route by the following prioritization:
+        // - The first route found that has upstream: varnish
+        // - if primary route has upstream: varnish, that route will be prioritised
+        // If no route is found with upstream: varnish, then purge_server will not be set
+        $route = null;
+        foreach ($routes as $host => $info) {
+            if ($route === null && $info['type'] === 'upstream' && $info['upstream'] === 'varnish') {
+                $route = $host;
+            }
+            if ($info['type'] === 'upstream' && $info['upstream'] === 'varnish' && $info['primary'] === true) {
+                $route = $host;
+                break;
+            }
+        }
+
+        if ($route !== null && !($_SERVER['HTTPCACHE_PURGE_TYPE'] ?? false)) {
+            $purgeServer = rtrim($route, '/');
+            if (($_SERVER['HTTPCACHE_USERNAME'] ?? false) && ($_SERVER['HTTPCACHE_PASSWORD'] ?? false)) {
+                $domain = parse_url($purgeServer, \PHP_URL_HOST);
+                $credentials = urlencode($_SERVER['HTTPCACHE_USERNAME']) . ':' . urlencode($_SERVER['HTTPCACHE_PASSWORD']);
+                $purgeServer = str_replace($domain, $credentials . '@' . $domain, $purgeServer);
+            }
+
+            $container->setParameter('purge_type', 'varnish');
+            $container->setParameter('purge_server', $purgeServer);
+        }
+        // Setting default value for HTTPCACHE_VARNISH_INVALIDATE_TOKEN if it is not explicitly set
+        if (!($_SERVER['HTTPCACHE_VARNISH_INVALIDATE_TOKEN'] ?? false)) {
+            $container->setParameter('varnish_invalidate_token', $_SERVER['PLATFORM_PROJECT_ENTROPY']);
+        }
+
+        // Adapt config based on enabled PHP extensions
+        // Get imagine to use imagick if enabled, to avoid using php memory for image conversions
+        if (\extension_loaded('imagick')) {
+            $container->setParameter('liip_imagine_driver', 'imagick');
+        }
```

Comment:
Optionally what I would do here is split this into smaller task-related methods for: dfs, redis cache, solr, http cache.